ddocs/medic-client/views/messages_by_contact_date/reduce.js

```diff
@@ -1,28 +1,17 @@
 function(key, values) {
-  var max = { date: 0 };
-  var read;
+  var latest = { date: 0 };
   values.forEach(function(value) {
-    if (!read || !value.read) {
-      read = value.read || [];
-    } else {
-      read = read.filter(function(user) {
-        return value.read.indexOf(user) !== -1;
-      });
-    }
-    if (value.date > max.date) {
-      max = value;
+    if (value.date > latest.date) {
+      latest = value;
     }
   });
-  max.read = read;
-
-  if (max.message) {
-    var code = max.message.charCodeAt(99);
+  if (latest.message) {
+    var code = latest.message.charCodeAt(99);
     if (0xD800 <= code && code <= 0xDBFF) {
```

Comment:
Cleaned up and commented - it's detecting if the character at position 99 is the first half of a unicode pair so we don't return half a smiley face...